common/djangoapps/enrollment/views.py

```diff
@@ -469,6 +472,14 @@ def get(self, request):
                     ).format(username=username)
                 }
             )
+        if username == request.user.username or self.has_api_key_permissions(request):
+            return Response(enrollment_data)
+        filtered_data = []
+        for enrollment in enrollment_data:
```

Comment:
@jimabramson @dsjen I don't think I'm hitting the database multiple times.  The [`has_user()` method](https://github.com/edx/edx-platform/blob/master/common/djangoapps/student/roles.py#L133-L146) caches all roles a user has on the Django user object the first time I test a role for that user, and all subsequent queries should be answered from that cache.

I've implemented the suggestion to check for `GlobalStaff` first.